plugins/CoreHome/angularjs/common/services/piwik.js

```diff
@@ -61,6 +65,22 @@
                 return false;
             }
         }
+
+        function updateDateInTitle( date, period ) {
+            piwikApi = piwikApi || $injector.get('piwikApi');
+            piwikApi.fetch(
+                {
+                    module: 'CoreHome',
+                    action: 'getDateString',
+                    date: date,
+                    period: period,
+                }
+            ).then(function (response) {
+                if (response && response.date) {
+                    document.title = piwik.siteName + ' - ' + response.date + ' - ' + _pk_translate('CoreHome_WebAnalyticsReports') + ' - Piwik';
```

Comment:
We cannot hardcode "Piwik" here (which should be "Matomo" btw), because when for example a custom logo is used, we do not show "Matomo" in the title. 

I don't know how this can be solved as it'll be hard to detect which part of the title needs to be replaced with a title. And the title may be different to the title specified here. 

I wouldn't even mind if we just removed the date from the title in general.

Reply:
Thinking about it... I reckon if we still wanted to have the date in the page title, I would say lets only set the date in JS and not in PHP/Twig at all. This will be easiest and allows us to change the title etc. and have logic only in one place. So JS could append the date to whatever title is currently set, and when date is changed, remove the last part and update it again. Would make it a lot simpler and more consistent.

Reply:
I really like the idea of de-duplicating this logic. It would be simple to cache the title that was generated on the back-end, so no tomfoolery is needed when updating the title again (just append the new date string to the cached title). Shall I make this change in this PR?

Reply:
This PR or another PR 👍